app/main.py

```diff
@@ -65,14 +65,23 @@ def custom_openapi():
     return app.openapi_schema
 app.openapi = custom_openapi
 
-def valid_query_id(query_id: str):
+def query_id_validator(query_id: str):
     if not query_id: return False
     # AWS Athena query ID format is a md5 hash with 4 hyphen
     if ( query_id.count('-') != 4 ): return False
     # length of query id = 32 (md5 hash length) + 4 (hyphen)
     if ( len(query_id) != 36 ): return False
     return True
 
+def cache_key_generator(data_type: str, species: str, fields: str = "", condition: str = ""):
+    # lowercase all SQL keywords as it has no significance in identifying a unique request
+    condition = condition.replace("AND", "and").replace("BETWEEN", "between").replace("LIKE", "like").replace("IN", "in")
```

Comment:
Will lower casing the whole `condition` variable cause any issues?
I was thinking of the case where users provide gene IDs as `ENSG00000210049` or `ensg00000210049` for example.
This will generate two different query IDs even though the queries are technically the same.

Reply:
@bilalebi The values in the queries are case sensitive, attached an example for reference
![Screenshot 2022-08-11 153327](https://user-images.githubusercontent.com/17514193/184110868-b6fde67e-008e-4c0c-82ec-c51d7bf69c7e.png)
![Screenshot 2022-08-11 153253](https://user-images.githubusercontent.com/17514193/184110876-174e35ce-f3bf-41b3-9677-2284be1db7d7.png)
e

Reply:
Interesting, thanks for the clarification, it's good to keep this in mind